packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -417,6 +417,9 @@ def _get_provider(self, provider_id: Optional[str]) -> BaseProvider:
 
         return self.providers[provider_id]
 
+    def _is_chat_model(self, provider_id: str) -> bool:
+        return provider_id in ["anthropic-chat", "bedrock-chat"]
+
```

Comment:
Instead of maintaining a hardcoded list of chat models, how about we make this a computed property on `BaseProvider`? For example:

```py
from langchain.chat_models.base import BaseChatModel

...

class BaseProvider(BaseModel):
  ...

  @property
  def _is_chat_provider(self):
    """Whether this provider inherits from a LangChain `BaseChatModel`."""
    return isinstance(self, BaseChatModel)
```